text/0085-api-keys.md

```diff
@@ -0,0 +1,548 @@
+- Title: API Keys
+- Start Date: 2021-10-15
+- Specification PR: [#85](https://github.com/meilisearch/specifications/pull/85)
+- Discovery Issue: [#51](https://github.com/meilisearch/product/issues/51)
+
+# API Keys
+
+## 1. Functional Specification
+
+### 1.1 Summary
+
+Granular management of API keys is added to MeiliSearch. It is possible to restrict the access of an API key to certain actions on specific indexes.
+
+### 1.2 Motivation
+
+To make MeiliSearch more reliable for teams, we extend the management and the possibilities of restrictions for the management of a MeiliSearch instance by introducing a concrete API reséource (`API Key`). Security is a critical need, often tricky to negotiate as the stakes are high for a company.
+
+### 1.3 Glossary
+
+| Term               | Definition |
+|--------------------|------------|
+| Master Key         | This is the master key that allows you to create other API keys. The master key is defined by the user when launching MeiliSearch, thus gives access to the `/keys` API endpoint. |
+| API Key            | API keys are stored and managed from the endpoint `/keys` by the master key holder. These are the keys used by the technical teams to interact with MeiliSearch at the level of the client code. |
+
+### 1.4 Personas
+
+| Persona | Role |
+|---------|------|
+| Anna    | Anna has an `ssh` access to the MeiliSearch instance and manages it on a daily basis. |
+| Mark    | Mark is a developer from the same company as Anna. He will implement the code to communicate with MeiliSearch to solve technical/product needs. |
+
+### 1.5 `API Key` Explanations
+
+#### 1.5.1 Summary Key Points
+
+- `X-MEILI-API-KEY` header is replaced by the `Authorization` header. `API keys` must be specified with the bearer authorization method. See examples.
+- `/keys` management is restricted to the master key.
+- When a master key is set at MeiliSearch first-launch, we generate two pre-configured default `API Key` resources. A `Default Search API Key` restricted to the search action and a `Default Admin API Key` to handle all operations (except managing `/keys` resource) on MeiliSearch.
+- If the master-key changes, all `API Keys` are re-generated.
+- These default API Keys can be modified/deleted with the `/keys` endpoint but are not re-created if MeiliSearch has already generated them at some point.
+- New endpoints are added to manage the `API Key` resource.
+- `API keys` can have restrictions on which methods can be accessed via an `actions` list; they can also `expiresAt` a specific date time and be restricted to a specific set of `indexes`.
+- There is no possibility to regenerate the value of the `key` field for a specific `API key` in this first iteration.
```

Comment:
Not sure to get what it means. What do you mean by "no possibility to regenerate the value"? You mean you have to create a new key, not to ask literally for a regeneration?

Reply:
Absolutely @curquiza. This is a future possibility!